packages/klighd-core/src/proxy-view/proxy-view.tsx

```diff
@@ -0,0 +1,742 @@
+                    const numProxiesInCurr = (transform as any).numProxies ?? 1;
+
+                    numProxiesInCluster += numProxiesInCurr;
+                    x += transform.x * numProxiesInCurr;
+                    y += transform.y * numProxiesInCurr;
+                }
+                x /= numProxiesInCluster;
+                y /= numProxiesInCluster;
+
+                // Make sure the calculated positions don't leave the canvas bounds
+                x = Math.max(0, Math.min(canvasWidth - size, x));
+                y = Math.max(0, Math.min(canvasHeight - size, y));
+                // Also make sure the calculated positions are still capped to the border (no floating proxies)
+                if (y > 0 && y < canvasHeight - size && (x < canvasWidth - size || x > 0)) {
+                    x = x > (canvasWidth - size) / 2 ? canvasWidth - size : 0;
+                }
+                if (x > 0 && x < canvasWidth - size && (y < canvasHeight - size || y > 0)) {
+                    y = y > (canvasHeight - size) / 2 ? canvasHeight - size : 0;
+                }
+
+                const clusterNode = this.getClusterRendering(`cluster-${i}-proxy`, numProxiesInCluster, size, x, y);
+                res.push({
+                    node: clusterNode,
+                    transform: {
+                        x: x, y: y, scale: 1, width: size, height: size,
+                        numProxies: numProxiesInCluster // Store the number of proxies in this cluster in case the cluster is clustered later on
+                    } as any as TransformAttributes
+                });
+            }
+
+            // Filter all overlapping nodes
+            res = res.filter((_, index) => !overlapIndexGroups.reduce((acc, group) => acc.concat(group), []).includes(index));
+        }
+
+        return res;
+    }
+
+    /** Returns the proxy rendering for a single off-screen node and applies logic, e.g. the proxy's position. */
+    private createSingleProxy(node: SKNode | VNode, transform: TransformAttributes, ctx: SKGraphModelRenderer): VNode | undefined {
+        let transformString = `translate(${transform.x}, ${transform.y})`;
+        if (transform.scale) {
+            transformString += ` scale(${transform.scale})`;
+        }
+
+        if (!(node instanceof SKNode)) {
+            // VNode, this is a predefined rendering (e.g. cluster)
+            this.updateTransform(node, transformString);
+            return node;
+        }
+
+        // Get VNode
+        const id = this.getProxyId(node.id);
+        let vnode = this.renderings.get(id);
+        if (!vnode) {
+            // Node hasn't been rendered yet (cache empty)
+
+            // Clone the node
+            const clone: SKNode = Object.create(node);
+            // Change its id for good measure
+            clone.id = id;
+            // Clear children, proxies don't show nested nodes (but keep Labels)
```

Comment:
what about ports?

Reply:
After giving it some thought I decided to also remove ports